xbmc/storage/MediaManager.cpp

```diff
@@ -760,3 +764,38 @@ void CMediaManager::RemoveDiscInfo(const std::string& devicePath)
   if (it != m_mapDiscInfo.end())
     m_mapDiscInfo.erase(it);
 }
+
+bool CMediaManager::playStubFile(const CFileItem& item)
+{
+  // Figure out Lines 1 and 2 of the dialog
+  std::string strLine1, strLine2;
+  CXBMCTinyXML discStubXML;
+  if (discStubXML.LoadFile(item.GetPath()))
+  {
+    TiXmlElement* pRootElement = discStubXML.RootElement();
+    if (!pRootElement || StringUtils::CompareNoCase(pRootElement->Value(), "discstub") != 0)
+      CLog::Log(LOGERROR, "Error loading %s, no <discstub> node", item.GetPath().c_str());
+    else
+    {
+      XMLUtils::GetString(pRootElement, "title", strLine1);
+      XMLUtils::GetString(pRootElement, "message", strLine2);
+    }
+  }
+
+  // Use the label for Line 1 if not defined
+  if (strLine1.empty())
+    strLine1 = item.GetLabel();
+
+  if (HasOpticalDrive())
+  {
+#ifdef HAS_DVD_DRIVE
+    if(CGUIDialogPlayEject::ShowAndGetInput(strLine1, strLine2))
+      return MEDIA_DETECT::CAutorun::PlayDiscAskResume();
+#endif
+  }
+  else
+  {
+    HELPERS::ShowOKDialogText(strLine1, strLine2);
```

Comment:
KODI::MESSAGING::HELPERS::ShowOKDialogText(strLine1, strLine2);

and remove the using namespace KODI::MESSAGING.

let someone else chime in in case they have a strong preference i guess though

Reply:
updated to this, i think it's a bit cleaner this way